### PR TITLE
[CP-3128] Flashing Harmony on MacOS

### PR DESCRIPTION
**Pull request by MateuszMudita**
JIRA Reference: [CP-3128]

### :memo: Description ️

-

### :muscle: Checklist before requesting a review - nice to have

- getState function in async thunk actions is correctly typed
- redux selectors are used in components / prop drilling is reduce

### :exclamation: Checklist before merging a pull request

- change went through the QA process
- translations are updated in dedicated application
- [CHANGELOG.md](./CHANGELOG.md) is updated

[CP-3128]: https://appnroll.atlassian.net/browse/CP-3128?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/msc-flash/msc-flash-harmony/src/lib/services/device-flash/macos/macos-device-flash-service.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    scriptPath: string
+  ): Promise<void> {
+    // const scriptPath = "./static/macflash.sh"
+    console.log("Script Path: ", scriptPath)
+    console.log("Image Path: ", imagePath)
+    console.log("Device: ", device)
+    console.log(
+      "Command: ",
+      `osascript -e 'tell application "Terminal" to do script "\\"${scriptPath}\\" -i \\"${imagePath}\\" -d \\"${device}\\""'`
+    )
```
Comment (dkarski) [inline libs/msc-flash/msc-flash-harmony/src/lib/services/device-flash/macos/macos-device-flash-service.ts:72]:
```suggestion
```

File: libs/msc-flash/msc-flash-harmony/src/lib/services/flash-msc-device.service.ts [hunk 2]
```diff
@@ -26,10 +26,9 @@ export const flashMscDeviceService =
       const flashingFiles = getState().flashing.mscFlashingFilesDetails
 
       if (flashingFiles) {
-        await downloadFlashingFiles(dispatch, flashingFiles)
-        await unpackFlashingImage(dispatch, flashingFiles)
+        // await downloadFlashingFiles(dispatch, flashingFiles)
+        // await unpackFlashingImage(dispatch, flashingFiles)
         await startFlashingProcess(dispatch, flashingFiles)
```
Comment (dkarski) [inline libs/msc-flash/msc-flash-harmony/src/lib/services/flash-msc-device.service.ts:31]:
```suggestion
        await downloadFlashingFiles(dispatch, flashingFiles)
        await unpackFlashingImage(dispatch, flashingFiles)
        await startFlashingProcess(dispatch, flashingFiles)
```

File: libs/msc-flash/msc-flash-harmony/src/lib/services/flash-msc-device.service.ts [hunk 3]
```diff
@@ -26,10 +26,9 @@ export const flashMscDeviceService =
...
-        await removeDownloadedMscFiles()
```
Comment (dkarski) [inline libs/msc-flash/msc-flash-harmony/src/lib/services/flash-msc-device.service.ts:32]: 😉

Comment (MateuszMudita) [reply libs/msc-flash/msc-flash-harmony/src/lib/services/flash-msc-device.service.ts:32]: I've moved it to `startFlashingProcess()` function :)

**Review by mkurczewski — APPROVED** [review]

**Pull request merged**